iset.mm

```diff
@@ -98005,6 +98021,113 @@ This definition (in terms of ` U. ` and ` ~<_ ` ) is not taken directly
     syl wceq sizeinf adantl breqtrrd ) ACDZEBFGZHAIJZKBIJZLUCUEKLGZUDUCUEMDUGUC
     UEANOUEPRQUDUFKSUCBTUAUB $.
 
+  $( A finite set is equinumerous to the range of integers from one up to the
+     hash value of the set.  (Contributed by Jim Kingdon, 22-Feb-2022.) $)
+  isfinite4im $p |- ( A e. Fin -> ( 1 ... ( size ` A ) ) ~~ A ) $=
+    ( cfn wcel c1 csize cfv cfz co wceq cen wbr cn0 sizecl sizefz1 syl wb cz 1z
+    nn0zd fzfig sylancr sizeen mpancom mpbid ) ABCZDAEFZGHZEFUFIZUGAJKZUEUFLCUH
+    AMZUFNOUGBCZUEUHUIPUEDQCUFQCUKRUEUFUJSDUFTUAUGAUBUCUD $.
+
+  $( Two ways of saying a finite set is empty.  (Contributed by Jim Kingdon,
+     22-Feb-2022.) $)
+  sizeeq0 $p |- ( A e. Fin -> ( ( size ` A ) = 0 <-> A = (/) ) ) $=
+    ( cfn wcel csize cfv c0 wceq cen wbr cc0 wb 0fin sizeen mpan2 c1 cfz fveq2i
+    co fz10 cn0 0nn0 sizefz1 ax-mp eqtr3i eqeq2i en0 3bitr3g ) ABCZADEZFDEZGZAF
+    HIZUIJGAFGUHFBCUKULKLAFMNUJJUIOJPRZDEZUJJUMFDSQJTCUNJGUAJUBUCUDUEAUFUG $.
+
+  $( Two ways of saying a finite set is not empty.  Also, "A is inhabited"
+     would be equivalent by ~ fin0 .  (Contributed by Jim Kingdon,
+     23-Feb-2022.) $)
+  sizeneq0 $p |- ( A e. Fin -> ( 0 < ( size ` A ) <-> A =/= (/) ) ) $=
+    ( cfn csize cfv cc0 cap wbr wne clt c0 cz wb sizecl nn0zd 0zd zapne syl2anc
+    wcel cn0 cr cle nn0re nn0ge0 ap0gt0 syl sizeeq0 necon3bid 3bitr3d ) ABRZACD
+    ZEFGZUJEHZEUJIGZAJHUIUJKREKRUKULLUIUJAMZNUIOUJEPQUIUJSRZUKUMLZUNUOUJTREUJUA
+    GUPUJUBUJUCUJUDQUEUIUJEAJAUFUGUH $.
+
+  $( Positive natural closure of the hash function.  (Contributed by Jim
+     Kingdon, 23-Feb-2022.) $)
+  sizenncl $p |- ( A e. Fin -> ( ( size ` A ) e. NN <-> A =/= (/) ) ) $=
+    ( cfn wcel csize cfv cn c0 wne wb cc0 wceq wa simpr nnne0 sizeeq0 necon3bid
+    adantl adantr mpbid wn 2thd necon2bi biimpa syl 2falsed cn0 wo sizecl elnn0
+    nner sylib mpjaodan ) ABCZADEZFCZUOAGHZIUNJKZUMUOLZUOUPUMUOMURUNJHZUPUOUSUM
+    UNNZQUMUSUPIUOUMUNJAGAOZPRSUAUMUQLZUOUPUQUOTUMUOUNJUTUBQVBAGKZUPTUMUQVCVAUC
+    AGUJUDUEUMUNUFCUOUQUGAUHUNUIUKUL $.
+
+  $( The empty set has size zero.  (Contributed by Jim Kingdon,
+     23-Feb-2022.) $)
+  size0 $p |- ( size ` (/) ) = 0 $=
+    ( c0 csize cfv cc0 wceq eqid cfn wcel wb 0fin sizeeq0 ax-mp mpbir ) ABCDEZA
+    AEZAFAGHNOIJAKLM $.
+
+  $( The size of a singleton.  (Contributed by Jim Kingdon, 23-Feb-2022.) $)
```

Comment:
"... is 1." ?

Reply:
I think I tend to like the terser comments in a case like this, where the theorem itself is so clear. Not that I have an especially strong opinion on it.